Mage/src/main/java/mage/cards/ModalDoubleFacedCard.java

```diff
@@ -116,7 +116,7 @@ private void setSideZones(Zone mainZone, Game game) {
     @Override
     public boolean moveToZone(Zone toZone, Ability source, Game game, boolean flag, List<UUID> appliedEffects) {
         if (super.moveToZone(toZone, source, game, flag, appliedEffects)) {
-            setSideZones(toZone, game);
+            setSideZones(game.getState().getZone(getId()), game);
```

Comment:
it’s shared logic for all double faced cards. Must fix same way all other (both methods):

1.
![IMG_0256](https://github.com/magefree/mage/assets/8344157/bf5a0e33-a50b-4edc-ad33-cc58949604b8)

2.
![IMG_0255](https://github.com/magefree/mage/assets/8344157/6cbf3ea0-6dc2-47d9-b8d2-67c1cf0b0412)